pynicotine/pynicotine.py

```diff
@@ -219,16 +219,17 @@ def __init__(self, bindip, port):
 
     def start(self, ui_callback=None, network_callback=None):
 
-        log.add(_("Loading %(program)s %(version)s"), {"program": "Python", "version": config.python_version})
-        log.add(_("Loading %(program)s %(version)s"), {"program": config.application_name, "version": config.version})
-        log.add_debug("Using Python executable: %s", str(sys.executable))
-
         self.ui_callback = ui_callback
         self.network_callback = network_callback if network_callback else self.network_event
-
         script_dir = os.path.dirname(__file__)
-        self.geoip = GeoIP(os.path.join(script_dir, "geoip/ipcountrydb.bin"))
 
+        log.add(_("Loading %(program)s %(version)s"), {"program": "Python", "version": config.python_version})
+        log.add_debug("Using %(program)s executable: %(exe)s", {"program": "Python", "exe": str(sys.executable)})
+        log.add_debug("Using %(program)s executable: %(exe)s", {"program": config.application_name, "exe": script_dir})
+        log.add_debug("Using configuration: %(file)s", {"file": config.filename})
```

Comment:
Perhaps include this in load_config in config.py instead.

Reply:
Perhaps importing `log` into `config` might be problematic if there was some fundamental runtime problem.

I suppose if we have got this far through the program execution then we would hope we can load the log module, but maybe this isn't certain in all cases.